core/src/test/scala/integration/kafka/server/ReplicaManagerTest.scala

```diff
@@ -0,0 +1,61 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package integration.kafka.server
+
+import kafka.server.KafkaConfig
+import kafka.utils.TestUtils
+import kafka.zk.ZooKeeperTestHarness
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.utils.MockTime
+import org.junit.Test
+import org.junit.Assert._
+
+class ReplicaManagerTest extends ZooKeeperTestHarness {
+  @Test
+  def testMaybePropagateIsrChanges(): Unit = {
+    val config = KafkaConfig.fromProps(TestUtils.createBrokerConfig(1, zkConnect))
+    val mockTime = new MockTime(10000)
```

Comment:
Do we need to set autoTick in MockTime? I thought we don't want the scheduler in replicaManager to trigger maybePropagateIsrChanges().

Reply:
maybePropagateIsrChanges() won't do anything unless ReplicaManager.IsrChangePropagationInterval milliseconds have elapsed since the last call to recordIsrChange.